### PR TITLE
Vacuum and Dump fail on big repository (XP 6.15) #8069

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/elasticsearch/executor/ScrollExecutor.java  2
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/RepoDumper.java  1
         

Complexity decreasing per file
==============================
+ modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/vacuum/versiontable/VersionTableCleanupTask.java  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5534126&bid=18141292)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/elasticsearch/executor/ScrollExecutor.java  1
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/RepoDumper.java  1
         

Complexity decreasing per file
==============================
+ modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/vacuum/versiontable/VersionTableCleanupTask.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5534126&bid=18141292)

File: modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/RepoDumper.java [hunk 1]
```diff
@@ -2,13 +2,15 @@
...
+import java.util.HashSet;
 import java.util.Set;
+import java.util.function.Consumer;
 import java.util.stream.Collectors;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.google.common.collect.Sets;
+import jdk.net.SocketFlow;
```
Comment (rymsha) [inline modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/RepoDumper.java:13]: should not be here

File: modules/core/core-api/src/main/java/com/enonic/xp/node/AbstractQuery.java [hunk 2]
```diff
@@ -44,6 +45,8 @@
 
     private final boolean explain;
 
+    private final Consumer<Object> batchCallback;
```
[the remark this answers is not in the document]
Comment (rymsha) [reply modules/core/core-api/src/main/java/com/enonic/xp/node/AbstractQuery.java:48]: Or, for now remove generic `<Object>`

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/RepoDumper.java  1
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/elasticsearch/executor/ScrollExecutor.java  1
         

Complexity decreasing per file
==============================
+ modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/vacuum/versiontable/VersionTableCleanupTask.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5534126&bid=18141292)

File: modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/vacuum/versiontable/VersionTableCleanupTask.java [hunk 3]
```diff
@@ -181,28 +147,56 @@ private boolean versionUsedInAnyBranch( final Repository repository, final NodeV
...
+
+            }
+            this.repositoryCleanStarted = true;
         }
 
-        final RangeFilter mustBeOlderThanFilter = RangeFilter.create().
-            fieldName( VersionIndexPath.TIMESTAMP.getPath() ).
-            to( ValueFactory.newDateTime( this.until ) ).
-            build();
+        if ( searchResult.getHits().getSize() == 0 )
```
Comment (anatol-sialitski) [inline modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/vacuum/versiontable/VersionTableCleanupTask.java:182]: will be better to use `isEmpty`

Comment (vbradnitski) [reply modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/vacuum/versiontable/VersionTableCleanupTask.java:182]: There is no such method in `SearchHits` class. Not sure that is need to be added just for one call.

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/elasticsearch/executor/ScrollExecutor.java  1
- modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/RepoDumper.java  1
         

Complexity decreasing per file
==============================
+ modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/vacuum/versiontable/VersionTableCleanupTask.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5534126&bid=18141292)

**Pull request merged**